app-backend/src/main/kotlin/org/example/appbackend/config/JwtTokenFilter.kt

```diff
@@ -0,0 +1,103 @@
+import org.example.appbackend.service.UserService
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.web.filter.OncePerRequestFilter
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.security.Keys
+import jakarta.servlet.FilterChain
+import jakarta.servlet.ServletException
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import java.io.IOException
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jws
+import io.jsonwebtoken.JwtException
+import java.security.Key
+import java.util.Date
+import java.util.*
+import org.slf4j.LoggerFactory
+
+class JwtTokenFilter(
+        @Value("\${spring.jwt.secret}")
+        private val jwtSecret: String,
+        private val userService: UserService
+) : OncePerRequestFilter() {
+
+    private var logger = LoggerFactory.getLogger(JwtTokenFilter::class.java)
+
+    @Throws(ServletException::class, IOException::class)
+    override fun doFilterInternal(
+            request: HttpServletRequest,
+            response: HttpServletResponse,
+            filterChain: FilterChain
+    ) {
+        val authorizationHeader = request.getHeader("Authorization")
+
+        var userId: Int? = null
+        var jwtToken: String?
+
+        if (authorizationHeader != null && authorizationHeader.startsWith("Bearer ")) {
+            jwtToken = authorizationHeader.substring(7)
+            logger.info("/JwtTokenFilter: jwtToken= $jwtToken")
+            try {
+                val session = request.session
+                val token = session.getAttribute("token") as? String
+                logger.info("/JwtTokenFilter: session token = $token")
+                if (token != jwtToken) {
+                    throw IllegalArgumentException("Token not found.")
+                }
+                userId = extractUserId(jwtToken)
+                logger.info("/JwtTokenFilter: userId = $userId")
+            } catch (e: IllegalArgumentException) {
+                // Log the error
+                logger.error("Invalid JWT token: ${e.message}")
+                // Return HTTP unauthorized response
+                response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Invalid JWT token")
+                return
+            }
+        }
+
+        if (userId != null && SecurityContextHolder.getContext().authentication == null) {
+            val userDetails = userService.getUserById(userId)
+            logger.info("/JwtTokenFilter: username= ${userDetails.username}")
+            if (userDetails != null) {
+                val authenticationToken = UsernamePasswordAuthenticationToken(
+                        userDetails, null, null
+                )
+                authenticationToken.details = WebAuthenticationDetailsSource().buildDetails(request)
+                SecurityContextHolder.getContext().authentication = authenticationToken
+                logger.info("/JwtTokenFilter: end.")
```

Comment:
quite strange end logging because there are lots of paths that will not reach this statement, but finish the function in some way. Maybe it should be "validation successful" or anything else

Reply:
fixed.